src/buildercore/trop.py

```diff
@@ -94,44 +96,58 @@ def rds_security(context):
 #
 
 
-def instance_tags(context):
-    return [
-        ec2.Tag('Name', context['stackname']),
+def instance_tags(context, node=None):
+    tags = [
         ec2.Tag('Owner', context['author']),
+        # hierarchically ordered
         ec2.Tag('Project', context['project_name']),
+        ec2.Tag('Cluster', context['stackname']),
     ]
-
-def ec2instance(context):
+    if node:
+        tags.append(ec2.Tag('Name', '%s--%d' % (context['stackname'], node)))
+        tags.append(ec2.Tag('Node', '%s' % node))
+    else:
+        tags.append(ec2.Tag('Name', context['stackname']))
+    return tags
+
+def ec2instance(context, node):
     lu = partial(utils.lu, context)
+    build_vars = dict(context)
+    build_vars['node'] = node
+    build_vars['nodename'] = "%s--%s" % (context['stackname'], node)
+    # the above context will reside on the server at /etc/build-vars.json.b64
+    # this gives Salt all (most) of the data that was available at template compile time.
+    build_vars_serialization = base64.b64encode(json.dumps(build_vars))
```

Comment:
there was a large if for adding build vars to context only in the case of EC2 instances being there. Moving the serialization to base64 here avoids adding to `context` a `build_vars` key containing the encoding of the whole `context` again. Also, it's naturally executed only on EC2 instances given the function we are in.